packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2606,7 +2606,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      */
     private verifyNotClosed() {
         if (this._disposed) {
-            throw new Error("Runtime is closed");
+            throw new UsageError("Runtime is closed");
```

Comment:
I don't know if i agree with this. UsageError has generally meant that the end developer or consumer called something incorrectly, maybe with the wrong argument. However, this case could happen from anywhere.

Reply:
Context is highly important in a case like this, i don't think this call site knows enough to classify its own error

Reply:
to avoid over classification of dataprocessing error, i wonder if we should only create dataprocessing errors when things are not disposed, as if the runtime is disposed, we generally are no longer processing data

Reply:
That aside, what error type would you suggest for this error?  I don't think throwing un-errorType'd errors from our codebase is desirable.

Reply:
i disagree. i don't think we can or should classify all errors at their creation point. the appropriated classify errors needs context, and code like this doesn't have it. in these cases some layer above with classify it if requried, otherwise the container will just mark it as generic error

Reply:
Ok, yeah interesting. I'll close this for now, it's fine for this to be DataProcessingError.  Whatever we can do to help partners be more independent when it comes to investigating them, the better; in this case we got pinged since this comes from our code.  The error message is pretty self-explanatory, so at least there's that.

Reply:
i do like the idea of no longer creating data processing errors after we shut down, as once things are closed, we aren't really processing anything for real anymore

Reply:
#9872

Reply:
another option could also be to just not log any more errors after the container is closed, but someone would have to think about it more to figure out if that is actually a good idea.

Reply:
My gut says that would come back to bite us

Reply:
yeah. i guess we could miss stuff, or cause problems after close that we would then be blind to